omigami/spec2vec.py

```diff
@@ -90,30 +90,39 @@ def _build_payload(
         - payload: JSON
             the full request payload with input data
         """
-        spectra = []
-        for spectrum in batch:
-            spectra.append(
-                {
-                    "peaks_json": str(
-                        [
-                            [mz, intensity]
-                            for mz, intensity in zip(
-                                spectrum.peaks.mz, spectrum.peaks.intensities
-                            )
-                        ]
-                    ),
-                    "Precursor_MZ": str(spectrum.metadata["pepmass"][0]),
-                }
-            )
+        spectra_payload = []
+        for spectrum in spectra:
+            spectrum_payload = {
+                "peaks_json": str(
+                    [
+                        [mz, intensity]
+                        for mz, intensity in zip(
+                            spectrum.peaks.mz, spectrum.peaks.intensities
+                        )
+                    ]
+                )
+            }
+            try:
+                spectrum_payload["Precursor_MZ"] = str(spectrum.metadata["pepmass"][0])
+            except KeyError:
+                if "precursor_mz" not in spectrum.metadata:
+                    raise KeyError(
+                        "One of 'precursor_mz' or 'pepmass' must be present for every spectra"
+                    )
+                spectrum_payload["Precursor_MZ"] = str(
+                    spectrum.metadata["precursor_mz"][0]
+                )
```

Comment:
Instead of catching an exception and using it as control flow, I would rather check if `pepmass` key exists in `metadata` keys. We shouldn't "expect" that an exception might be raised and carry on.

Reply:
```suggestion
            metadata_keys = spectrum.metadata.keys()
            if "pepmass" in metadata_keys:
               spectrum_payload["Precursor_MZ"] = str(spectrum.metadata["pepmass"][0])
            elif "precursor_mz" in metadata_keys:
               # fallback, we use precursor_mz in case pepmass is not present. Check issue #20 for background.
               spectrum_payload["Precursor_MZ"] = str(
                    spectrum.metadata["precursor_mz"][0]
                )
             else:
                 raise KeyError(
                     "Either 'precursor_mz' or 'pepmass' must be present in every spectra."
                 )        
```